### DISCUSSION TITLE
Troubleshooting Mindcode errors

### DISCUSSION THREAD

**Discussion by cardillan** [cardillan/mindcode#157] [General]
Up to now, encountering a syntax error in Mindcode was pretty challenging. The upcoming release, 2.3.0, should provide much better error messages. In any case, if you run into trouble with compiling Mindcode and don't feel like opening a new issue for it, you can ask for help here.

**Comment by stoltzld** [comment]
I tried pasting the following schematic in the encoded schematic text area at https://mindcode.herokuapp.com/decompiler?s=a0b9eb77-7a9f-4c32-be29-e63e80d2b1d8 and got the error "info.teksol.schemacode.SchematicsInternalError: Invalid plan index -1 for unit factory @air-factory."  I'm assuming the error is because I didn't choose for my air factory to produce anything in my schematic.

bXNjaAF4nCVQy07DMBCcPJs2Lb0ghMQlP2CJ9g8Qd07cEAc3WYpV167spKifAXwUH8XDrBvJ2vHMzs5axgLLDLmRe8LNbXNvHTUPQ6tp8M16tWseV2Ld3CmHuiPfOnXolTUASi03pD3Sp+c5iqMcdI9pa00vlSGH5capbkuClSOdrMO85WRhxmRctna/GXzMElviAdmz50p2nerVkYQjHvS9G9qoV4PRVnYcW0vlxIuM6glLr7Rin/B70j13853SBhdbJw+vqidxcOQ9Zgf7Rk4Y2xG/+xpIUfAFCapYEswiS7GIJUU2CSF8hvD3FclIP84047lpNOWj+n5Wweo5r4gJ4NBiBGY5QzZumkRxGncVXBJkJTPuTVAjTzgtZBU3Sv5Zhip66wglAw/NGGrwmaOsshCQhr/wHX7CL/4Bge1zcw==

**Reply by cardillan** [reply]
Thanks for reporting the issue! Yes, it does seem the problem is with not setting up the air factory. I'll fix it in the next release, which should be out soon.

**Reply by cardillan** [reply]
Turns out the fix wasn't complete: the result of decompiling the schematics in Mindcode 3.1 contains this block:

```
schematic
    @air-factory         at (10,  3) facing south unit 
end
```

which doesn't build, because the `unit` keyword is superfluous without the unit plan.

The (hopefully final) fix will be part of Mindustry 3.2.
